src/overlays/actors/ovl_En_Fg/z_en_fg.c

```diff
@@ -19,42 +41,417 @@ const ActorInit En_Fg_InitVars = {
+                Audio_PlayActorSound2(this, 0x28B1);
+                func_80A2D280(&this->skelAnime, 3);
+                this->actor.velocity.y = 10.0f;
+                this->timer = Rand_S16Offset(30, 30);
+                this->actionFunc = func_80A2D778;
+            }
+    }
+    Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fg_0x80A2D280/func_80A2D400.asm")
+void func_80A2D778(EnFg* this, GlobalContext* globalCtx) {
+    Actor* ac;
+    s32 pad;
+    s16 rotY;
+    s16 rotX;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fg_0x80A2D280/func_80A2D42C.asm")
+    switch (func_80A2D42C(this)) {
+        case 4:
+            this->actor.flags &= ~1;
+            this->skelAnime.animPlaybackSpeed = 0.0f;
+            ac = this->collider.base.ac;
+            rotY = ac->world.rot.y;
+            rotX = ac->world.rot.x;
+            this->actor.scale.x = fabsf(Math_CosS(rotY) * 0.01f);
+            this->actor.scale.y = fabsf(Math_CosS(rotX) * 0.01f);
+            this->actor.scale.z = fabsf(Math_SinS(rotY) * 0.01f);
+            this->actor.scale.x = CLAMP_MIN(this->actor.scale.x, 0.001f);
+            this->actor.scale.y = CLAMP_MIN(this->actor.scale.y, 0.001f);
+            this->actor.scale.z = CLAMP_MIN(this->actor.scale.z, 0.001f);
+            this->actionFunc = func_80A2D9CC;
+            break;
+        case 3:
+            break;
+        case 1:
+            this->actor.flags &= ~1;
+            Audio_PlayActorSound2(this, 0x28E3);
+            func_80A2D280(&this->skelAnime, 0);
+            this->actor.params = 5;
+            this->skelAnime.animPlaybackSpeed = 0.0f;
+            ac = this->collider.base.ac;
+            this->actor.world.rot.y = Math_Vec3f_Yaw(&ac->world.pos, &this->actor.world.pos);
+            this->actor.shape.rot = this->actor.world.rot;
+            this->actor.velocity.y = 10.0f;
+            this->actor.speedXZ = 3.0f;
+            this->actor.gravity = -0.8f;
+            this->unk_2FA = 1;
+            this->timer = 0;
+            this->actionFunc = func_80A2D9DC;
+            break;
+        default:
+            if (func_801378B8(&this->skelAnime, 8.0f)) {
+                this->skelAnime.animCurrentFrame = 8.0f;
+                this->skelAnime.animPlaybackSpeed = 0.0f;
+            }
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fg_0x80A2D280/func_80A2D4B8.asm")
+            if ((this->actor.velocity.y <= 0.0f) && (this->actor.bgCheckFlags & 1)) {
+                func_80A2D280(&this->skelAnime, 0);
+                this->actionFunc = func_80A2D4B8;
+                this->actor.velocity.y = 0.0f;
+            } else {
+                Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+            }
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Fg_0x80A2D280/func_80A2D778.asm")
+void func_80A2D9CC(EnFg* this, GlobalContext* globalCtx) {
```

Comment:
```suggestion
void EnFgDoNothing(EnFg* this, GlobalContext* globalCtx) {
```
I don't know if there were thoughts of documenting things yet?

Reply:
Kinda? This feels like an incomplete/beta file so I found it hard to document where the devs were going with certain functions. This function name makes sense though. I can give it another look

Reply:
I'm not yet at the stage of requiring documentation.
Since documentation wasn't really started, no need to review for it on this file.

Reply:
Sounds good. I did use this name suggestion and named one other function too